jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/AliasEndpoint.java

```diff
@@ -117,20 +123,43 @@ public Response crossOriginForInstallations(@Context HttpHeaders headers, @PathP
 	 *             required values).
 	 * @statuscode 401 The request requires authentication.
 	 *
-	 *             TODO - Rename to registered
 	 */
 	@GET
 	@Path("/exists/{alias}")
 	@Produces(MediaType.APPLICATION_JSON)
 	@ReturnType("java.lang.Boolean")
-	public Response registered(@PathParam("alias") String alias, @Context HttpServletRequest request) {
-		if (aliasService.registered(alias))
-			return appendAllowOriginHeader(Response.ok().entity(Boolean.TRUE), request);
+	@Deprecated
+	public Response registeredDeprecated(@PathParam("alias") String alias, @Context HttpServletRequest request) {
+		return isAliasRegistered(alias, request);
+	}
 
-		return appendAllowOriginHeader(Response.ok().entity(Boolean.FALSE), request);
+	/**
+	 * RESTful API for validating alias is already registered (Keycloak). The
+	 * Endpoint has public access.
+	 *
+	 * @param alias The alias name.
+	 * @return {@link Boolean}
+	 *
+	 * @responseheader Access-Control-Allow-Origin With host in your "Origin" header
+	 * @responseheader Access-Control-Allow-Credentials true
+	 * @responseheader WWW-Authenticate Basic realm="UnifiedPush Server" (only for
+	 *                 401 response)
+	 *
+	 * @statuscode 200 True/False String value.
+	 * @statuscode 400 The format of the aliases request was incorrect (e.g. missing
+	 *             required values).
+	 * @statuscode 401 The request requires authentication.*
+	 */
+	@GET
+	@Path("/registered/{alias}")
+	@Produces(MediaType.APPLICATION_JSON)
+	@ReturnType("java.lang.Boolean")
+	public Response registered(@PathParam("alias") String alias, @Context HttpServletRequest request) {
+		return isAliasRegistered(alias, request);
 	}
 
 	/**
+	 * @deprecated
```

Comment:
Can you please add comment explaining why is this API deprecated and which API should the client use instead?

Reply:
I added the @deprecated to documentation only, the @Deprecated annotation was there log before me...